### PR TITLE
Strip '\r' from the end of each line when parsing the vcpkg database

### PR DESCRIPTION
**Pull request by browntarik**
Addresses: https://github.com/microsoft/vscode-cpptools/issues/12413

We were incorrectly parsing our vcpkg header database when deciding to present the "Learn how to install a library for this header with vcpkg" code action.

The issue was due to not stripping the "\r" line endings that appear exclusively on Windows machines.

This fix will now strip the "\r\n" line endings on windows properly and if will only strip "\n" for linux/mac machines.

It should now display the following behavior on windows (see below):
Screenshot 2024-06-26 125711

### CODE REVIEW

File: Extension/src/LanguageServer/extension.ts [hunk 1]
```diff
@@ -106,7 +109,7 @@ async function lookupIncludeInVcpkg(document: vscode.TextDocument, line: number)
     if (!matches || !matches.length || !matches.groups) {
         return [];
     }
-    const missingHeader: string = matches.groups.includeFile.replace(/\//g, '\\');
+    const missingHeader: string = matches.groups.includeFile.replace(/[\\\/]/g, '\\');
```
Comment (bobbrow) [inline Extension/src/LanguageServer/extension.ts]: I don't think this line needs to change. Do you know what was the problem here?  You just added backslash to the list of things to replace, but you'd just replace it with a backslash so that doesn't change the string.

Comment (browntarik) [reply Extension/src/LanguageServer/extension.ts]: I see I thought the regex was slightly wrong, but I just misinterpreted how JavaScript applies backslashes.

File: Extension/src/LanguageServer/extension.ts [hunk 2]
```diff
@@ -58,7 +58,11 @@ async function initVcpkgDatabase(): Promise<VcpkgDatabase> {
         const zip = new StreamZip.async({ file: util.getExtensionFilePath('VCPkgHeadersDatabase.zip') });
         try {
             const data = await zip.entryData('VCPkgHeadersDatabase.txt');
-            const lines = data.toString().split('\n');
+            // Strip '\r' and '\n' from the end of each line if they exist.
+            let lines = data.toString().split('\r\n');
+            if (lines.length === 1) {
```
Comment (bobbrow) [inline Extension/src/LanguageServer/extension.ts]: Do we not know what the line endings are?  We should own this file and how it's formatted.  If there is a way to split with a regex, you could split on `\r?\n` instead.

File: Extension/src/LanguageServer/extension.ts (hunk 1 above)
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/extension.ts]: Why is this needed? Isn't the changed code converting `\\` with `\\`?

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]

File: Extension/src/LanguageServer/extension.ts (hunk 2 above)
Comment (sean-mcmanus) [reply Extension/src/LanguageServer/extension.ts]: There's a `splitLines` function in `Extension/src/common.ts`.

File: Extension/src/LanguageServer/extension.ts (hunk 1 above)
Comment (sean-mcmanus) [reply Extension/src/LanguageServer/extension.ts]: Oh, I didn't see Bob's comment.

File: Extension/src/LanguageServer/extension.ts [hunk 3]
```diff
@@ -58,6 +58,7 @@ async function initVcpkgDatabase(): Promise<VcpkgDatabase> {
         const zip = new StreamZip.async({ file: util.getExtensionFilePath('VCPkgHeadersDatabase.zip') });
         try {
             const data = await zip.entryData('VCPkgHeadersDatabase.txt');
+            // Strip '\r' and '\n' from the end of each line if they exist.
```
Comment (bobbrow) [inline Extension/src/LanguageServer/extension.ts]: This comment seems out of place now

**Review by sean-mcmanus — APPROVED** [review]
